core/src/typecheck/unif.rs

```diff
@@ -1162,18 +1345,59 @@ impl Unify for UnifEnumRows {
                     Err(RowUnifError::UnboundTypeVariable(id))
                 }
                 (EnumRowsF::Empty, EnumRowsF::Empty) => Ok(()),
-                (EnumRowsF::Empty, EnumRowsF::Extend { row: ident, .. }) => {
-                    Err(RowUnifError::ExtraRow(ident))
-                }
-                (EnumRowsF::Extend { row: ident, .. }, EnumRowsF::Empty) => {
-                    Err(RowUnifError::MissingRow(ident))
-                }
-                (EnumRowsF::Extend { row: id, tail }, erows2 @ EnumRowsF::Extend { .. }) => {
+                (
+                    EnumRowsF::Empty,
+                    EnumRowsF::Extend {
+                        row: UnifEnumRow { id, .. },
+                        ..
+                    },
+                ) => Err(RowUnifError::ExtraRow(id)),
+                (
+                    EnumRowsF::Extend {
+                        row: UnifEnumRow { id, .. },
+                        ..
+                    },
+                    EnumRowsF::Empty,
+                ) => Err(RowUnifError::MissingRow(id)),
+                (
+                    EnumRowsF::Extend {
+                        row: UnifEnumRow { id, typ },
+                        tail,
+                    },
+                    erows2 @ EnumRowsF::Extend { .. },
+                ) => {
                     let uerows2 = UnifEnumRows::Concrete {
                         erows: erows2,
                         var_levels_data: var_levels2,
                     };
-                    let (_, t2_without_row) = uerows2.remove_row(&id, state, ctxt.var_level)?;
+                    let (ty2_result, t2_without_row) =
+                        //TODO[adts]: it's ugly to create a temporary Option just to please the
+                        //Box/Nobox types, we should find a better signature for remove_row
```

Comment:
I may be missing something obvious, but where is the temporary option? Is this comment outdated?

Reply:
Sorry, I think I wrote that most for myself than for the reader :stuck_out_tongue: what I mean is that we have an `Option<Box<T>>`, and that we clone this whole thing and dereference the inner pointer to get `Option<T>`, while I think `remove_erows` might be better off taking an `Option<Box<_>>` to begin with. But I agree the text of the todo is a bit off.